### PR TITLE
feat: add VestingNFT

### PR DESCRIPTION
**Pull request by Kud8**

### CODE REVIEW

File: contracts/NFT/NftTimelock.sol [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    function nft() public view virtual returns (IERC721) {
+        return _nft;
+    }
+
+    /**
+     * @dev Returns the token ID of the NFT being held.
+     * Returns undefined if the contract is not holding an NFT.
+     */
+    function tokenId() public view virtual returns (uint256) {
+        return _tokenId;
```
Comment (lebed2045) [inline contracts/NFT/NftTimelock.sol:64]: i don't undersatnd how does it work. Where accounts are handled?

File: contracts/NFT/NftTimelock.sol [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        require(
+            nft().ownerOf(tokenId()) == address(this),
+            "BasicNftTimelock: no NFT to release"
+        );
+
+        // Transfer NFT to beneficiary
+        nft().safeTransferFrom(address(this), beneficiary(), tokenId());
+
+        // Check if beneficiary has received NFT, if not, revert
+        require(
```
Comment (lebed2045) [inline contracts/NFT/NftTimelock.sol:103]: is there reference of AAA companie's code who do that? seems redundant

File: contracts/NFT/VestingNft.sol [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+ * after 1 year".
+ */
+contract VestingNft is IVestingNft {
+    mapping(address => UserInfo) public userInfo;
+
+    function addNft(
+        IERC721 _nft,
+        uint256 _tokenId,
+        address _beneficiary,
+        uint256 _releaseTime
```
Comment (lebed2045) [inline contracts/NFT/VestingNft.sol]: _releaseTimestamp

File: contracts/interfaces/IVestingNft.sol [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    /**
+     * @notice Info of each user (beneficiary of token after they are released)
+     * `nft` ERC721 basic token smart contract
+     * `tokenId` ERC721 basic token ID of contract being held
+     * `releaseTime` Timestamp when token release is enabled. It is specified as a Unix timestamp (in seconds)
+    **/
+    struct UserInfo {
+        IERC721 nft;
+        uint256 tokenId;
+        uint256 releaseTime;
```
Comment (Kud8) [inline contracts/interfaces/IVestingNft.sol:17]: need to be named as "releaseTimestamp"

**Review by lebed2045 — COMMENTED** [review]
ship it!

**Pull request merged**